### PR TITLE
Properly list subworkflows in run workflow form

### PR DESCRIPTION
**Pull request by guerler**
This PR ensures that subworkflows are properly listed as steps in the run workflow form by relaxing the rendering condition. Previously only `data_input` and `data_collection_input` were shown in addition to `tool` modules.

### CODE REVIEW

**Comment by bgruening** [comment]
This was fast! Thanks for fixing!

File: client/galaxy/scripts/mvc/tool/tool-form-composite.js [hunk 1]
```diff
@@ -446,23 +446,8 @@ define([ 'utils/utils', 'utils/deferred', 'mvc/ui/ui-misc', 'mvc/form/form-view'
...
-                                                                         .attr( 'href', '/history/switch_to_history?hist_id=' + invocation.history.id )
-                                                                         .text( invocation.history.name ) ) );
-                    _.each( invocation.outputs, function( output ) {
-                        $invocation.append( $( '<div/>' ).addClass( 'messagerow' ).html( '<b>' + output.hid + '</b>: ' + output.name ) );
-                    });
-                    $message.append( $invocation );
-                }
-                return $message;
+                return $( '<div/>' ).addClass( 'donemessagelarge' )
+                                    .append( $( '<p/>' ).text( 'The requested workflows have been queued and datasets will appear as jobs are created - you will need to refresh your history panel to see these.' ) );
```
Comment (bgruening) [inline client/galaxy/scripts/mvc/tool/tool-form-composite.js]: I think this is a useful change and un-clutters the UI. Nevertheless, can we preserve the history-ids? Let's say something like `have been queued and datasets (5-45) will appear ...`

Comment (dannon) [inline client/galaxy/scripts/mvc/tool/tool-form-composite.js]: Hrmm.  Do we have a feeling for how many people will miss the "datasets will appear in a new history _link_to_new_history_"?  I feel like that was added due to user demand years ago.

Comment (guerler) [inline client/galaxy/scripts/mvc/tool/tool-form-composite.js]: I would have preferred to leave the dataset details in the success message but the existing api endpoint we are planning to use does not provide that information, nor does it allow to submit the datasets to a new history or supports batch modes. However, the plan is to enhance the existing endpoint and add at least the latter two features and then add the new_history_link back to the success message. I will transfer the changes and close this PR in favor of #2077.

**Comment by guerler** [comment]
I re-opened this issue since it is actually not related to the api endpoint discussion.

**Pull request merged**